tests/results/DiffTest.py

```diff
@@ -207,6 +209,164 @@ def test_from_string_arrays(self):
+        diff = ['--- a/testfile',
+                '+++ b/testfile',
+                '@@ -1,2 +1,1 @@',
+                ' A',
+                '-B',
+                '@@ -5,3 +4,2 @@',
+                ' E',
+                ' F',
+                '-G']
+        diff_string = '\n'.join(diff)
+        self.uut = Diff.from_unified_diff(diff_string, source)
+        self.assertEqual(self.uut.original, source)
+        self.assertEqual(self.uut.modified, target)
+
+        source = ['A', 'C', 'D', 'E', 'G']
+        target = ['A', 'B', 'C', 'D', 'E', 'F', 'G']
+        diff = ['--- a/testfile',
+                '+++ b/testfile',
+                '@@ -1,1 +1,2 @@',
+                ' A',
+                '+B',
+                '@@ -4,2 +5,3 @@',
+                ' E',
+                '+F',
+                ' G']
+        diff_string = '\n'.join(diff)
+        self.uut = Diff.from_unified_diff(diff_string, source)
+        self.assertEqual(self.uut.original, source)
+        self.assertEqual(self.uut.modified, target)
+
+        source = ['A', 'B', 'C', 'D', 'E', 'F', 'G']
+        target = ['A', 'B', 'Z', 'D', 'E', 'F', 'K']
+        diff = ['--- a/testfile',
+                '+++ b/testfile',
+                '@@ -1,3 +1,3 @@',
+                ' A',
+                ' B',
+                '-C',
+                '+Z',
+                '@@ -6,2 +5,2 @@',
+                ' F',
+                '-G',
+                '+K']
+        diff_string = '\n'.join(diff)
+        self.uut = Diff.from_unified_diff(diff_string, source)
+        self.assertEqual(self.uut.original, source)
+        self.assertEqual(self.uut.modified, target)
+
+        # test invalid unified diffs
+        source = ['first', 'second']
+        target = ['only_first_changed', 'second']
+        diff = ['--- a/testfile',
+                '+++ b/testfile',
+                '@@ -1,2 +1,2 @@',
+                '-line_to_be_deleted_is_not_same',
+                '+only_first_changed',
+                ' second']
+        diff_string = '\n'.join(diff)
+        with self.assertRaises(RuntimeError):
+            self.uut = Diff.from_unified_diff(diff_string, source)
+
+        source = ['first', 'second']
+        target = ['first', 'second']
+        diff = ['--- a/testfile',
+                '+++ b/testfile',
+                '@@ -1,2 +1,2 @@',
+                ' context_line_is_not_same',
+                ' second']
+        diff_string = '\n'.join(diff)
+        with self.assertRaises(RuntimeError):
+            self.uut = Diff.from_unified_diff(diff_string, source)
+
+        source = ['first', 'invalid starting character']
+        target = ['first', 'invalid starting character']
+        diff = ['--- a/testfile',
+                '+++ b/testfile',
+                '@@ -1,2 +1,2 @@',
+                ' first',
+                '*invalid_starting_character']
+        diff_string = '\n'.join(diff)
+        with self.assertRaises(UnidiffParseError):
```

Comment:
if you want to support proper errors, sure :+1:
Though you can request from the users to supply valid diffs, otherwise undefined behaviour. If you want :)